File: CSharpPractice/LeetcodePractice/Leetcode36.cs
public class Solution {
    public bool IsValidSudoku(char[][] board) {
        
        Dictionary<int,HashSet<int>> colSet  = new Dictionary<int,HashSet<int>>();
        Dictionary<int,HashSet<int>> rowSet  = new Dictionary<int,HashSet<int>>();
        Dictionary<int,HashSet<int>> boxIndexSet  = new Dictionary<int,HashSet<int>>();

        for (int i = 0; i < board.Length; i++)
        {
            for (int j = 0; j < board[0].Length; j++)
            {
                if (board[i][j]!='.')
                {
                    int theVal = board[i][j]-'0';
                    int boxIndex = 3*(i/3)+j/3;

                    if (!colSet.ContainsKey(j))
                    {
                        colSet[j] = new HashSet<int>();
                    }
                    if (!rowSet.ContainsKey(i))
                    {
                        rowSet[j] = new HashSet<int>();
                    }
                    if (!boxIndexSet.ContainsKey(boxIndex))
                    {
                        boxIndexSet[boxIndex] = new HashSet<int>();
                    }


                    bool isValid = !colSet[j].Contains(theVal)||!rowSet[i].Contains(theVal)||!boxIndexSet[boxIndex].Contains(theVal);
                    if (!isValid)
                    {
                        return false;
                    }
                    colSet[j].Add(theVal);
                    rowSet[i].Add(theVal);
                    boxIndexSet[boxIndex].Add(theVal);

                }
                
            }
            
        }

        return true;
        

    }
}

File: CppPractice/CppLeetcode/Leetcode36.cpp
#include<set>
#include<vector>
#include<utility>
using namespace std;

class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) 
    {
        set<pair<int,int>> colSet; // col,val
        set<pair<int,int>> rowSet; // row,val
        set<pair<int,int>> boxSet;

        for (int i = 0; i < board.size(); i++)
        {
            for (int j = 0; j < board[0].size(); j++)
            {
                if (board[i][j]!='.')
                {
                    int theVal = int(board[i][j]); 
                    int boxIndex = 3* (i/3) + j/3 ;
                    if (colSet.find({j,theVal})!=colSet.end() ||
                        rowSet.find({i,theVal})!=rowSet.end() ||
                        boxSet.find({boxIndex,theVal})!=boxSet.end())
                    {
                        return false;
                    }
                    colSet.insert({j,theVal});
                    rowSet.insert({i,theVal});
                    boxSet.insert({boxIndex,theVal});
                }
            }
        }
        return true;
    }
};